internal/pkg/agent/application/coordinator/config_patcher.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package coordinator
+
+import (
+	"context"
+)
+
+type ConfigPatch func(change ConfigChange) ConfigChange
+
+// ConfigPatchManager is a decorator to restore some agent settings from the elastic agent configuration file
+type ConfigPatchManager struct {
+	inner   ConfigManager
+	outCh   chan ConfigChange
+	patchFn ConfigPatch
+}
+
+func (c ConfigPatchManager) Run(ctx context.Context) error {
+	go c.patch(c.inner.Watch(), c.outCh)
```

Comment:
An unchecked goroutine here feels a bit brittle/unsafe. Could we end up with an explosion of running goroutines? Should there be a timeout?

Reply:
Yes. If somehow, `inner.Run()` returns right away, but `inner.Watch()` doesn't at all, then you could end up with a goroutines leak.
I also don't have much context on this code at all, so maybe that's not really possible.

Reply:
`inner.Watch()` returns the channel on which config changes are published. We use that channel to intercept configuration changes that we want to patch.
The basic pattern for a ConfigManager is to start it and then consume all the configuration changes passed on the channel. 
The ConfigPatchManager does the same thing with the addition of a goroutine that will read from the inner channel, patch and publish on its own channel